src/Uno.UI/DataBinding/BindingPropertyHelper.cs

```diff
@@ -1242,37 +1240,159 @@ private static DependencyProperty FindDependencyProperty(Type ownerType, string
 					{
 						return fastConvertResult;
 					}
-					else if (t != typeof(object))
+					else if (t is null || t == typeof(object))
+					{
+						return value;
+					}
+					else if (t == typeof(string))
 					{
-						value = ConvertWithConvertionExtension(value, t);
+						return value?.ToString();
+					}
+					else if (t.IsEnum)
+					{
+						var valueString = value.ToString();
+
+						FieldInfo? defaultValue = null;
```

Comment:
Let's move those blocks to separate methods, one per if condition, this will make this method easier to read.